### ISSUE TITLE
[SelectableTile] Selected state is out of sync when user do shift+click for multi-selection

### ISSUE THREAD

**Issue by ZhiboMZhang** [carbon-design-system/carbon#7759]
The last selected item can lose selected state in certain scenario.
Steps to reproduce:
1) Build a view with list of selectable tile
2) Click on one tile and then hold shift to do multi-selection on tiles
3) Hold the shift key and select any tile again
Bug -> if the tile was previously selected, the tile is deselected, even though the `select` props passed in indicate that the tile should persistent selection
![holdingShiftToDeselectObjects](https://user-images.githubusercontent.com/14354863/107256086-09ffb380-6a07-11eb-8e90-8066adf87d56.gif)

Looked into the code base, the issue seems from here:
https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/Tile/Tile.js#L228
https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/Tile/Tile.js#L239-L247
When user click on the tile, the selected state doesn't respect the selected props value.

## Environment

> Operating system Mac OS

> Browser Chrome

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using? "carbon-components": "10.17.0", "carbon-components-react": "7.17.0",

> What did you expect to happen? The selected state should respect the selected props value passed in.

> What happened instead? The selected state automatically toggled when user click on the tile

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. Click on the fourth tile, it should select all the four tiles, but the last tile is toggled as deselected.
> Please create a reduced test case in CodeSandbox
> - React:
>   https://codesandbox.io/s/s-m4yeb

Thanks Bill for helping me put up this codesandbox link. The critical line is at line 76.

## Additional information

- Screenshots or code
![Feb-08-2021 15-19-46](https://user-images.githubusercontent.com/14354863/107276514-20b30400-6a21-11eb-8783-88661c8191b8.gif)

- Notes

**Comment by emyarod** [comment]
imo the tiles should either be fully controlled (like Carbon checkboxes currently) or fully uncontrolled, rather than a mixture of the two

**Comment by emyarod** [comment]
I believe that shift click behavior is not supported out of the box by the multi select tile though
